File: src/main/java/com/fic/udc/es/padel/services/FieldService.java
package com.fic.udc.es.padel.services;

import java.util.List;

import com.fic.udc.es.padel.model.entities.Field;
import com.fic.udc.es.padel.model.exceptions.InstanceNotFoundException;

public interface FieldService {
	
	public Field addField(String name);
	
	public void deleteField(Long id) throws InstanceNotFoundException;
	
	public Field updateField(Long id, String name) throws InstanceNotFoundException;
	
	public void changeState(Long id, boolean state) throws InstanceNotFoundException;
	
	public List<Field> findAllFields();
}

File: frontend/src/backend/fieldService.js
import {config, appFetch} from './appFetch';

export const getAllFields = (onSuccess) =>
    appFetch(`/fields`, config('GET'), onSuccess);

export const addField = (field, onSuccess, onErrors) =>
    appFetch('/fields', config('POST', field), 
            onSuccess, onErrors);

export const updateField = (field, onSuccess, onErrors) => {
    const fieldId = field["fieldId"];
    appFetch(`/fields/${fieldId}`, config('PUT', field), 
            onSuccess, onErrors);
}

export const deleteField = ({fieldId}, onSuccess) =>
    appFetch(`/fields/${fieldId}`, config('DELETE'), onSuccess);

export const changeState = (state, onSuccess) => {
    const fieldId = state["id"];
    appFetch(`/fields/${fieldId}/changeState`, config('PUT', state), onSuccess);
}
